src/parser/Parser.ts

```diff
@@ -661,16 +684,28 @@ export class Parser {
                 }
 
                 if (match(Lexeme.Else)) {
+                    //consume any trailing colons
+                    while (check(Lexeme.Colon)) { advance(); }
+
                     match(Lexeme.Newline);
                     elseBranch = block(Lexeme.EndIf);
                     advance(); // skip past "end if"
                     match(Lexeme.Newline);
                 } else {
                     match(Lexeme.Newline);
-                    endIf = consume(
+                    endIfToken = consume(
                         `Expected 'end if' to close 'if' statement started on line ${startingLine}`,
                         Lexeme.EndIf
                     );
+                    
+                    //ensure that single-line `if` statements have a colon right before 'end if'
+                    if (ifToken.location.start.line === endIfToken.location.start.line) {
```

Comment:
I'm detecting a missing colon by checking that the `if` token is on the same line as the `end if`. It feels a little weird comparing line numbers...but I couldn't find another way around it. Open to suggestions, but it works...

Reply:
Interesting!  I suspect moving this check and error message up to line 695 and checking for either a newline or a `:` would get the job done nice and cleanly, without having to compare line numbers directly :smiley: 

Probably something like this?
```typescript
if (match(Lexeme.Newline, Lexeme.Colon)) {
    while (match(Lexeme.Newline, Lexeme.Colon));
} else if (check(Lexeme.EndIf)) {
    addError(peek(), "Expected ':' or newline to preceed 'end if'");
}
```

Reply:
I just tried your proposal, and it didn't work. It failed a few of the tests. I think because this code is after the `elseif` block captures. The point of this code is to detect when the `endif` is on the SAME line as the `if`...and in the situation where the are newlines in any of the `elseif` sections, your proposed solution wouldn't detect that. 

As an alternative, I could inspect every token between the `if` and the `endif`, and throw an error if no colon and no newline were detected. **However**, that's more expensive than a simple line check...not sure it's worth the effort.

Reply:
I see!  Interestingly, I don't seem to be able to get into this branch at all - putting an `endif`-terminated `if` statement on one line with no `:` preceding the `end if` didn't report "Expected ':' or newline to precede 'end if'":

```brightscript
sub main()
    if true: print "foo": else: print "bar" end if
end sub

sub main2()
    if true: print "foo" end if
end sub
```

produces
```
$ brs scratch/newline.brs 
scratch/newline.brs(2,44-50): Found unexpected token 'end if'
scratch/newline.brs(3,0-7): Found unexpected token 'end sub'
scratch/newline.brs(6,25-31): Found unexpected token 'end if'
scratch/newline.brs(7,0-7): Found unexpected token 'end sub'
scratch/newline.brs(6,4-6): Expected 'end if', 'else if', or 'else' to terminate 'then' block
scratch/newline.brs(6,25-31): Found unexpected token 'end if'
scratch/newline.brs(8,0-1): (At end of file) Expected 'end sub' to terminate sub block
Error occurred parsing
```

Perhaps I'm missing something!

Reply:
So apparently I was wrong about the elseif blocks being processed. Here's a statement that produces the `Expected ':' to preceed 'end if'` error message.

```brightscript
function main()
    if 1 < 2 : return true end if
end function
```

Reply:
I see!  This works for now, but I have a gut feeling that the `block` function is going to learn some of these tricks pretty soon :wink: